components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -53,7 +54,8 @@ class DownloadsFeature(
      * to be processed.
      */
     override fun start() {
-        super.observeSelected()
+        val session = sessionId?.let { sessionManager.findSessionById(sessionId) }
+        session?.let { observeFixed(it) } ?: observeSelected()
```

Comment:
Since that is exactly the same code as in the other PR. Maybe that could just be a method on `SelectionAwareSessionObserver` (which takes an optional session id)?

Reply:
Heh, left this comment in the other PR. I'll file an issue to do that.